File: A4.Empower-Git/A4.Empower/A4.BAL/Recruitment/Application/InterviewScheduleModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace A4.BAL
{
    public class InterviewScheduleViewModel
    {
        public InterviewScheduleViewModel()
        {
            InterviewScheduleLevelList = new List<InterviewScheduleLevel>();
            ManagerList = new List<DropDownList>();
            InterviewTypeList = new List<DropDownList>();
        }
        public List<InterviewScheduleLevel> InterviewScheduleLevelList { get; set; }
        public List<DropDownList> InterviewTypeList { get; set; }
        public List<DropDownList> ManagerList { get; set; }
    }
    public class InterviewScheduleLevel
    {
        public InterviewScheduleLevel()
        {
            InterviewScheduleModelList = new List<InterviewScheduleModel>();
        }
        public string LevelId { get; set; }
        public string Level { get; set; }
        public bool IsLevelCompleted { get; set; }
        public string InterviewId { get; set; }
        public bool IsInterviewCompleted { get; set; }
        public List<string> LevelManagerIds { get; set; }
        public List<InterviewScheduleModel> InterviewScheduleModelList { get; set; }
    }
    public class InterviewScheduleModel
    {
        public string InterviewId { get; set; }

        public DateTime Date { get; set; }

        public string Time { get; set; }

        public string ManagerId { get; set; }

        public string InterviewTypeId { get; set; }

        public string ManagerName { get; set; }

        public string Comment { get; set; }

        public bool InterviewStatus { get; set; }


        public bool IsLevelCompleted { get; set; }

        public bool IsCandidateSelected { get; set; } 

    }
     
    
}

File: ClientApp/src/app/models/recruitment/candidate-view/interview-schedule.model.ts

export class InterViewSchedule {
    constructor(id?: string, candidateName?: string, date?: Date, time?: string) {

        this.id = id;
        this.date = date;
        this.time = time;


    }
    public name: string;
    public jobInterviewTypeId: string
    public jobApplicationId:string;
    public interviewId: string;
    public interviewTypeId: string;
  public managerId: string;
  public jobCandidateUrl: string="";

    public id: string;
    public date: Date;
    public time: string;
    public interviewMode: string;
    public levelId : string;
    public isLevelCompleted:boolean;
    public managerIdList: string[];
}
